### PR TITLE
[CLD-5982] First commit of Mattermost Calls Offloader service Helm chart

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Mattermost Calls Offloader service Helm chart

### CODE REVIEW

File: charts/mattermost-calls-offloader/templates/hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v2beta1
```
Comment (angeloskyratzakos) [inline charts/mattermost-calls-offloader/templates/hpa.yaml:2]: I think there is v2 stable
https://github.com/kubernetes/api/tree/master/autoscaling

**Review by angeloskyratzakos — APPROVED** [review]
Nice work @stylianosrigas 🎉

File: charts/mattermost-calls-offloader/templates/hpa.yaml (hunk 1 above)
Comment (stylianosrigas) [reply charts/mattermost-calls-offloader/templates/hpa.yaml]: Nice catch @angeloskyratzakos thanks. Fixed ;)

File: charts/mattermost-calls-offloader/values.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  - name: LOGGER_ENABLEFILE
+    value: "false"
+  - name: JOBS_APITYPE
+    value: "kubernetes"
+  - name: JOBS_MAXCONCURRENTJOBS
+    value: "1"
+  - name: API_SECURITY_ALLOWSELFREGISTRATION # This should only be set to true if running the service inside a private network.
+    value: "true"
+  - name: LOGGER_CONSOLELEVEL
+    value: "DEBUG"
```
Comment (spirosoik) [inline charts/mattermost-calls-offloader/values.yaml:83]: I think as a default value we can add INFO here vs DEBUG

File: charts/mattermost-calls-offloader/templates/hpa.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.autoscaling.enabled }}
```
Comment (spirosoik) [inline charts/mattermost-calls-offloader/templates/hpa.yaml:1]: is this supposed to be supported out of the box?

File: charts/mattermost-calls-offloader/values.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
+  type: NodePort
```
Comment (spirosoik) [inline charts/mattermost-calls-offloader/values.yaml:40]: do we need NodePort? calls offloader works only in the internal network if I am not wrong.

**Review by spirosoik — COMMENTED** [review]
few minor comments

File: charts/mattermost-calls-offloader/values.yaml (hunk 2 above)
Comment (stylianosrigas) [reply charts/mattermost-calls-offloader/values.yaml]: Thanks fixed

File: charts/mattermost-calls-offloader/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v2
```
Comment (spirosoik) [inline charts/mattermost-calls-offloader/Chart.yaml:1]: wondering if it make sense to change the `calls` helm chart to have a dependency on this chart. It's not for now just an idea to streamline the installation of the calls in k8s.

**Review by spirosoik — APPROVED** [review]
thanks mate. great job, one minor comments which is not for now, just food for though.

File: charts/mattermost-calls-offloader/Chart.yaml (hunk 5 above)
Comment (streamer45) [reply charts/mattermost-calls-offloader/Chart.yaml:1]: This service enables functionalities which are still behind flags and default to off. In other words, the offloader isn't strictly needed for a working Calls deployment but only if someones wants to enable recordings (which require an enterprise license by the way). But again I am not sure what a dependency means in helm chart semantics.

**Review by streamer45 — APPROVED** [review]
Thanks! Looking forward to seeing this in action :rocket: 

I've also created https://mattermost.atlassian.net/browse/MM-53786 to update our official docs which I can take care of once this is merged and will ask for your review.

**Review by stafot — APPROVED** [review]
LGTM

File: charts/mattermost-calls-offloader/values.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 2
+
+image:
+  repository: mattermost/calls-offloader
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "v0.1.0"
```
[the remark this answers is not in the document]
Comment (stylianosrigas) [reply charts/mattermost-calls-offloader/values.yaml]: Updated to official v0.3.0

**Pull request merged**